src/services/accounts/AccountsValidateService.spec.ts

```diff
@@ -75,6 +90,9 @@ describe('Validate addresses', () => {
 		const expectedResponse = {
 			isValid: true,
 			ss58Prefix: '0',
+			network: 'polkadot',
+			accountId:
+				'0x002a39366f6620a6c2e2fed5990a3d419e6a19dd127fc7a50b515cf17e2dc5cc592312',
```

Comment:
OK, so I looked into it more, and all of these are incorrect. What you have here is `(prefix, AccountId, checksum)`, not `AccountId`. This breaks down to:

- Prefix: `00`
- AccountId: `2a39366f6620a6c2e2fed5990a3d419e6a19dd127fc7a50b515cf17e2dc5cc59`
- Checksum: `2312`

For example, in subkey:

```bash
> subkey inspect --public 0x002a39366f6620a6c2e2fed5990a3d419e6a19dd127fc7a50b515cf17e2dc5cc592312
Error: Input("Failed to construct public key from given hex")
```

But strip away the prefix and checksum:

```bash
> subkey inspect --public 0x2a39366f6620a6c2e2fed5990a3d419e6a19dd127fc7a50b515cf17e2dc5cc59 -n polkadot
Network ID/version: polkadot
  Public key (hex):   0x2a39366f6620a6c2e2fed5990a3d419e6a19dd127fc7a50b515cf17e2dc5cc59
  Account ID:         0x2a39366f6620a6c2e2fed5990a3d419e6a19dd127fc7a50b515cf17e2dc5cc59
  Public key (SS58):  1xN1Q5eKQmS5AzASdjt6R6sHF76611vKR4PFpFjy1kXau4m
  SS58 Address:       1xN1Q5eKQmS5AzASdjt6R6sHF76611vKR4PFpFjy1kXau4m
```

Reply:
This also aligns with the `AccountId`s in the previous tests.

Reply:
Thank you @joepetrowski for the super valuable feedback!

I changed the code so that if the user calls this endpoint with an address of : 
- valid length and
- in hex (from a u8 array) format
now I print an account id stripped from the prefix + checksum. 

Since I strip characters from specific indexes, I was wondering if there is any case where the user can input an address of another hex format of valid length ? If yes and if this format does not have the prefix + checksum in the same indexes then this solution might not work.

Reply:
Abstractly, yes, it's possible. If you look at [MultiAddress](https://github.com/paritytech/substrate/blob/6a59cc3/primitives/runtime/src/multiaddress.rs#L26) it does support `Raw(Vec<u8>),`. Practically speaking, most `AccountId` stuff uses this [32 byte](https://github.com/paritytech/substrate/blob/6a59cc3/primitives/core/src/crypto.rs#L483) object and converts [pub keys into it](https://github.com/paritytech/substrate/blob/6a59cc3/primitives/runtime/src/lib.rs#L328L335).